waspc/data/Generator/templates/react-app/src/webSocket.ts

```diff
@@ -0,0 +1,76 @@
+{{={= =}=}}
+
+import { useState, useEffect } from 'react'
+import { io, Socket } from 'socket.io-client'
+import { DefaultEventsMap, EventsMap } from '@socket.io/component-emitter'
+
+import { getAuthToken, events as apiEvents, EventType } from './api'
+import config from './config'
+
+// TODO: In the future, it would be nice if users could pass more
+// options to `io`, likely via some `configFn`.
+export const socket = io(config.apiUrl, { autoConnect: {= autoConnect =} })
+
+function refreshAuthToken() {  
+  // NOTE: When we figure out how `auth: true` works for Operations, we should
+  // mirror that behavior here for WebSockets. Ref: https://github.com/wasp-lang/wasp/issues/1133
+  socket.auth = {
+    token: getAuthToken()
+  }
+
+  if (socket.connected) {
+    socket.disconnect()
+    socket.connect()
+  }
+}
+
+refreshAuthToken()
+apiEvents.on(EventType.SET_AUTH_TOKEN, refreshAuthToken)
+apiEvents.on(EventType.CLEAR_AUTH_TOKEN, refreshAuthToken)
+
+export function useSocket<
```

Comment:
I still think we can achieve the following (maybe now, maybe in v2):
- this file imports the user defined types from the server file (like we did for operations)
- it glues these types to this fn and the user gets full stack type safety

IMHO, if we have it for operations, it would be a missed opportunity not to do it here.

I can attempt to do this and I'll make a commit if it works well.

Reply:
I think it is worth giving it a try, for sure!